fe/fe-core/src/main/java/org/apache/doris/nereids/rules/mv/SelectMaterializedIndexWithAggregate.java

```diff
@@ -711,34 +746,62 @@ private static Expression rewrite(Expression expr, RewriteContext context) {
 
         /**
          * count(distinct col) -> bitmap_union_count(mv_bitmap_union_col)
+         * count(col) -> sum(mv_count_col)
          */
         @Override
         public Expression visitCount(Count count, RewriteContext context) {
             if (count.isDistinct() && count.arity() == 1) {
```

Comment:
should the AggFuncRewriter process children first in each visit function? such as
```
public Expression visitCount(Count count, RewriteContext context) {
    List<Expression> children = count.children().stream().map(e -> this.rewrite(e, context)).collect(Collectors.toList());
    ...
}
```

Reply:
This rewriter is a bit different. 
It only handles aggregate function on a slot, e.g., rewriting `count(col)` to `sum(mv_count_col)`. The child of the aggregate function is checked by `ExpressionUtils.extractSlotOrCastOnSlot(count.child(0))` to ensure it's a slot.
 So we don't need to rewrite children first.